Author: Fatih Aşıcı <fatih@pardus.org.tr>
Description: n/a

diff -Naurp broadcom-sta.orig/amd64/src/wl/sys/wl_iw.h broadcom-sta/amd64/src/wl/sys/wl_iw.h
--- broadcom-sta.orig/amd64/src/wl/sys/wl_iw.h	2009-02-09 22:43:26.000000000 +0100
+++ broadcom-sta/amd64/src/wl/sys/wl_iw.h	2009-04-06 00:52:01.000000000 +0200
@@ -66,7 +66,7 @@ extern int wl_iw_ioctl(struct net_device
 extern void wl_iw_event(struct net_device *dev, wl_event_msg_t *e, void* data);
 extern int wl_iw_get_wireless_stats(struct net_device *dev, struct iw_statistics *wstats);
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 27)
+#if 1
 #define IWE_STREAM_ADD_EVENT(info, stream, ends, iwe, extra) \
 	iwe_stream_add_event(info, stream, ends, iwe, extra)
 #define IWE_STREAM_ADD_VALUE(info, event, value, ends, iwe, event_len) \
diff -Naurp broadcom-sta.orig/i386/src/wl/sys/wl_iw.h broadcom-sta/i386/src/wl/sys/wl_iw.h
--- broadcom-sta.orig/i386/src/wl/sys/wl_iw.h	2009-02-09 22:43:26.000000000 +0100
+++ broadcom-sta/i386/src/wl/sys/wl_iw.h	2009-04-06 00:52:05.000000000 +0200
@@ -66,7 +66,7 @@ extern int wl_iw_ioctl(struct net_device
 extern void wl_iw_event(struct net_device *dev, wl_event_msg_t *e, void* data);
 extern int wl_iw_get_wireless_stats(struct net_device *dev, struct iw_statistics *wstats);
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 27)
+#if 1
 #define IWE_STREAM_ADD_EVENT(info, stream, ends, iwe, extra) \
 	iwe_stream_add_event(info, stream, ends, iwe, extra)
 #define IWE_STREAM_ADD_VALUE(info, event, value, ends, iwe, event_len) \
