--- src/src/wl/sys/wl_linux.c
+++ src/src/wl/sys/wl_linux.c
@@ -214,9 +214,27 @@
 };
 #endif 
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)
+static const struct net_device_ops wlNetAdpNetDevOps =
+{
+    .ndo_open = wl_open,
+    .ndo_stop = wl_close,
+    .ndo_start_xmit = wl_start,
+    .ndo_get_stats = wl_get_stats,
+    .ndo_set_mac_address = wl_set_mac_address,
+    .ndo_set_multicast_list = wl_set_multicast_list,
+    .ndo_do_ioctl = wl_ioctl,
+};
+
 static
 void wl_if_setup(struct net_device *dev)
 {
+    dev->netdev_ops = &wlNetAdpNetDevOps;
+#else
+static
+void wl_if_setup(struct net_device *dev)
+{
+
 	dev->open = wl_open;
 	dev->stop = wl_close;
 	dev->hard_start_xmit = wl_start;
@@ -224,6 +242,7 @@
 	dev->set_mac_address = wl_set_mac_address;
 	dev->set_multicast_list = wl_set_multicast_list;
 	dev->do_ioctl = wl_ioctl;
+#endif
 #ifdef CONFIG_WIRELESS_EXT
 #if WIRELESS_EXT < 19
 	dev->get_wireless_stats = wl_get_wireless_stats;
only in patch2:
unchanged:
--- src/src/wl/sys/wl_iw.c
+++ src/src/wl/sys/wl_iw.c
@@ -67,6 +67,12 @@
 #define IW_EVENT_IDX(cmd)	((cmd) - IWEVFIRST)
 #endif 
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)
+#define DO_IOCTL(dev) dev->netdev_ops->ndo_do_ioctl
+#else
+#define DO_IOCTL(dev) dev->do_ioctl
+#endif
+
 static void swap_key_from_BE(
 	        wl_wsec_key_t *key
 )
@@ -118,7 +124,7 @@
 
 	fs = get_fs();
 	set_fs(get_ds());
-	ret = dev->do_ioctl(dev, &ifr, SIOCDEVPRIVATE);
+	ret = DO_IOCTL(dev)(dev, &ifr, SIOCDEVPRIVATE);
 	set_fs(fs);
 
 	return ret;
