diff -Nur -x '*.orig' -x '*~' bcmwl-5.10.91.9+bdcom/src/wl/sys/wl_linux.c bcmwl-5.10.91.9+bdcom.new/src/wl/sys/wl_linux.c
--- bcmwl-5.10.91.9+bdcom/src/wl/sys/wl_linux.c	2009-04-22 13:48:59.000000000 -0400
+++ bcmwl-5.10.91.9+bdcom.new/src/wl/sys/wl_linux.c	2009-10-22 23:27:41.276879918 -0400
@@ -51,6 +51,7 @@
 #include <linux/ethtool.h>
 #include <linux/completion.h>
 #include <linux/pci_ids.h>
+#include <linux/sched.h>
 #define WLC_MAXBSSCFG		1	
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 14)
