Index: broadcom-sta-5.100.82.111/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-5.100.82.111.orig/amd64/src/wl/sys/wl_linux.c	2011-10-06 03:16:38.000000000 +0400
+++ broadcom-sta-5.100.82.111/amd64/src/wl/sys/wl_linux.c	2011-10-25 08:27:33.248356736 +0400
@@ -185,6 +185,8 @@
 static void wl_report_radio_state(wl_info_t *wl);
 #endif
 
+MODULE_LICENSE("MIXED/Proprietary");
+
 static struct pci_device_id wl_id_table[] = {
 	{ PCI_VENDOR_ID_BROADCOM, 0x4311, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, 
 	{ PCI_VENDOR_ID_BROADCOM, 0x4312, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, 
Index: broadcom-sta-5.100.82.111/i386/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-5.100.82.111.orig/i386/src/wl/sys/wl_linux.c	2011-10-06 03:16:10.000000000 +0400
+++ broadcom-sta-5.100.82.111/i386/src/wl/sys/wl_linux.c	2011-10-25 08:27:33.248356736 +0400
@@ -185,6 +185,8 @@
 static void wl_report_radio_state(wl_info_t *wl);
 #endif
 
+MODULE_LICENSE("MIXED/Proprietary");
+
 static struct pci_device_id wl_id_table[] = {
 	{ PCI_VENDOR_ID_BROADCOM, 0x4311, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, 
 	{ PCI_VENDOR_ID_BROADCOM, 0x4312, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, 
