Index: broadcom-sta/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta.orig/amd64/src/wl/sys/wl_linux.c
+++ broadcom-sta/amd64/src/wl/sys/wl_linux.c
@@ -158,6 +158,8 @@ static void wl_report_radio_state(wl_inf
 
 MODULE_LICENSE("MIXED/Proprietary");
 
+MODULE_LICENSE("MIXED/Proprietary");
+
 static struct pci_device_id wl_id_table[] =
 {
 	{ PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
