Index: broadcom-sta-5.100.82.112/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-5.100.82.112.orig/amd64/src/wl/sys/wl_linux.c	2013-09-16 09:44:06.526230910 +0400
+++ broadcom-sta-5.100.82.112/amd64/src/wl/sys/wl_linux.c	2013-09-16 09:45:09.650231133 +0400
@@ -179,6 +179,8 @@
 static void wl_report_radio_state(wl_info_t *wl);
 #endif
 
+MODULE_LICENSE("MIXED/Proprietary");
+
 static struct pci_device_id wl_id_table[] =
 {
 	{ PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
