Index: broadcom-sta-5.100.82.112/i386/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-5.100.82.112.orig/i386/src/wl/sys/wl_linux.c	2011-11-08 19:54:08.000000000 +0100
+++ broadcom-sta-5.100.82.112/i386/src/wl/sys/wl_linux.c	2011-11-08 20:11:40.000000000 +0100
@@ -258,7 +256,7 @@
 #define quote_str(s) to_str(s)
 
 #ifndef BRCM_WLAN_IFNAME
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 #endif
 
 static char name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
Index: broadcom-sta-5.100.82.112/i386/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-5.100.82.112.orig/amd64/src/wl/sys/wl_linux.c	2011-11-08 19:54:08.000000000 +0100
+++ broadcom-sta-5.100.82.112/amd64/src/wl/sys/wl_linux.c	2011-11-08 20:11:40.000000000 +0100
@@ -258,7 +256,7 @@
 #define quote_str(s) to_str(s)
 
 #ifndef BRCM_WLAN_IFNAME
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 #endif
 
 static char name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
