Description: Allow normal users to send ioctl requests
Author: Cyril Lacoux <clacoux@easter-eggs.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=672677
Last-Update: 2012-11-19

Index: broadcom-sta-6.30.223.248/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-6.30.223.248.orig/amd64/src/wl/sys/wl_linux.c	2014-08-11 08:40:11.563128462 +0400
+++ broadcom-sta-6.30.223.248/amd64/src/wl/sys/wl_linux.c	2014-08-11 08:40:11.563128462 +0400
@@ -1653,11 +1653,7 @@
 	}
 
 	WL_LOCK(wl);
-	if (!capable(CAP_NET_ADMIN)) {
-		bcmerror = BCME_EPERM;
-	} else {
-		bcmerror = wlc_ioctl(wl->wlc, ioc.cmd, buf, ioc.len, wlif->wlcif);
-	}
+	bcmerror = wlc_ioctl(wl->wlc, ioc.cmd, buf, ioc.len, wlif->wlcif);
 	WL_UNLOCK(wl);
 
 done1:
