Index: broadcom-sta-6.30.223.248/amd64/src/wl/sys/wl_linux.c
===================================================================
--- broadcom-sta-6.30.223.248.orig/amd64/src/wl/sys/wl_linux.c	2014-10-20 09:55:20.147653836 +0400
+++ broadcom-sta-6.30.223.248/amd64/src/wl/sys/wl_linux.c	2014-10-20 10:00:36.499645692 +0400
@@ -1309,7 +1309,11 @@
 	dev->priv = priv_link;
 #else
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 17, 0)
+	dev = alloc_netdev(sizeof(priv_link_t), intf_name, NET_NAME_UNKNOWN, ether_setup);
+#else
 	dev = alloc_netdev(sizeof(priv_link_t), intf_name, ether_setup);
+#endif
 	if (!dev) {
 		WL_ERROR(("wl%d: %s: alloc_netdev failed\n",
 			(wl->pub)?wl->pub->unit:wlif->subunit, __FUNCTION__));
