diff --git a/src/Makefile b/src/Makefile
index 2ef91b1..6d821a0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -26,6 +26,12 @@
 #
 # $Id: Makefile_kbuild_portsrc,v 1.1.2.4 2008/05/02 18:33:39 Exp $
 
+ifeq ($(shell dpkg-architecture -qDEB_HOST_ARCH),amd64)
+BINARCH = x86_64
+else
+BINARCH = i386
+endif
+
 obj-m              += wl.o
 
 wl-objs            := 
@@ -37,4 +43,4 @@ EXTRA_CFLAGS       :=
 EXTRA_CFLAGS       += -I$(src)/src/include
 EXTRA_CFLAGS       += -I$(src)/src/wl/sys
 
-EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
+EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped_$(BINARCH)
